scipy/stats/tests/test_distributions.py

```diff
@@ -3067,64 +3100,121 @@ def test_pdf_nolan_samples(self):
+
+        # fmt: off
         tests = [
-            # best selects
-            ['best', None, 8, None],
-
-            # quadrature is accurate for most alpha except 0.25; perhaps
-            # limitation of Nolan stablec?
-            # we reduce size of x to speed up computation as numerical
-            # integration slow.
-            ['quadrature', None, 8,
-             lambda r: ((r['alpha'] > 0.25) &
-                        (npisin(r['x'], [-10, -5, 0, 5, 10])))],
-
-            # zolatarev is accurate except at alpha==1, beta != 0
-            ['zolotarev', None, 8, lambda r: r['alpha'] != 1],
-            ['zolotarev', None, 8,
-             lambda r: (r['alpha'] == 1) & (r['beta'] == 0)],
-            ['zolotarev', None, 1,
-             lambda r: (r['alpha'] == 1) & (r['beta'] != 0)],
-
-            # fft accuracy reduces as alpha decreases, fails at low values of
-            # alpha and x=0
-            ['fft', 0, 4, lambda r: r['alpha'] > 1],
-            ['fft', 0, 3, lambda r: (r['alpha'] < 1) & (r['alpha'] > 0.25)],
-            # not useful here
-            ['fft', 0, 1, lambda r: (r['alpha'] == 0.25) & (r['x'] != 0)],
+            # TODO: reduce range of pct to speed up computation as
+            # numerical integration slow, eg [.01, .05, .5, .95, .99]
+            ['dni', 1e-7, lambda r: ~(
+                ((r['beta'] == 0) & (r['pct'] == 0.5))
+                | ((r['beta'] >= 0.9) & (r['alpha'] >= 1.6)
+                    & (r['pct'] == 0.5))
+                | ((r['alpha'] <= 0.4) & npisin(r['pct'], [.01, .99]))
+                | ((r['alpha'] <= 0.3) & npisin(r['pct'], [.05, .95]))
+                | ((r['alpha'] <= 0.2) & npisin(r['pct'], [.1, .9]))
+                | ((r['alpha'] == 0.1) & npisin(r['pct'], [.25, .75])
+                    & npisin(np.abs(r['beta']), [.5, .6, .7]))
+                | ((r['alpha'] == 0.1) & npisin(r['pct'], [.5])
+                    & npisin(np.abs(r['beta']), [.1]))
+                | ((r['alpha'] == 0.1) & npisin(r['pct'], [.35, .65])
+                    & npisin(np.abs(r['beta']), [-.4, -.3, .3, .4, .5]))
+                | ((r['alpha'] == 0.2) & (r['beta'] == 0.5)
+                    & (r['pct'] == 0.25))
+                | ((r['alpha'] == 0.2) & (r['beta'] == -0.3)
+                    & (r['pct'] == 0.65))
+                | ((r['alpha'] == 0.2) & (r['beta'] == 0.3)
+                    & (r['pct'] == 0.35))
+                | ((r['alpha'] == 1.) & npisin(r['pct'], [.5])
+                    & npisin(np.abs(r['beta']), [.1, .2, .3, .4]))
+                | ((r['alpha'] == 1.) & npisin(r['pct'], [.35, .65])
+                    & npisin(np.abs(r['beta']), [.8, .9, 1.]))
+                | ((r['alpha'] == 1.) & npisin(r['pct'], [.01, .99])
+                    & npisin(np.abs(r['beta']), [-.1, .1]))
+                # various points ok but too sparse to list
+                | ((r['alpha'] >= 1.1))
+            )],
```

Comment:
What's the rationale behind this complicated set of excluded cases when testing the DNI method? For what it's worth, the test still passes on my machine when no filter is included like in the suggestion below. I'm trying to refactor the tests so that a small set of cases will be run in normal tests, a larger set in slow tests, and the full set in xslow tests. I wanted to try to keep the tests exactly the same for the full set, but this part is confusing to me.
```suggestion
            ['dni', 1e-7, None],
```

Reply:
IIRCC it's because it has to pass on multiple platforms (Windows, Linux, Mac, 32/64 bit). Most of these issues came out during the Scipy CI tests. It's also likely there are some redundant tests after we refactored the integrand for DNI using symmetry arguments etc.

Reply:
I see, that makes sense. I'd like to try to handle cases like that with conditional xfail pytest marks. I'll give some more thought to the best way to organize these.